packages/unstake/src/broker.rs

```diff
@@ -128,10 +128,16 @@ impl Broker {
 
     /// Commits the offer, deducts the reserve allocation from the total reservce, and returns
     /// messages that will instantiate the delegate contract with the debt tokens and ask tokens
-    pub fn accept_offer<T: CustomQuery>(&self, deps: DepsMut<T>, offer: &Offer) -> StdResult<()> {
+    pub fn accept_offer<T: CustomQuery>(
+        &self,
+        deps: DepsMut<T>,
+        offer: &Offer,
+    ) -> Result<(), ContractError> {
         let (mut available_reserve, mut deployed) = RESERVES.load(deps.storage).unwrap_or_default();
-        available_reserve -= offer.reserve_allocation;
-        deployed += offer.reserve_allocation;
+        available_reserve = available_reserve
+            .checked_sub(offer.reserve_allocation)
+            .map_err(|_| ContractError::InsufficentReserves {})?;
+        deployed = deployed.checked_add(offer.reserve_allocation)?;
```

Comment:
This may panic if `available_reserve` is less than `reserve_allocation`

Obviously, should never happen since offers are created and accepted atomically, but just in case